python/ray/rllib/es/policies.py

```diff
@@ -5,191 +5,85 @@
-            # arrays).
-            f.attrs['name'] = type(self).__name__
-            f.attrs['args_and_kwargs'] = np.void(pickle.dumps((self.args,
-                                                               self.kwargs),
-                                                              protocol=-1))
-
-    @classmethod
-    def Load(cls, filename, extra_kwargs=None):
-        with h5py.File(filename, 'r') as f:
-            args, kwargs = pickle.loads(f.attrs['args_and_kwargs'].tostring())
-            if extra_kwargs:
-                kwargs.update(extra_kwargs)
-            policy = cls(*args, **kwargs)
-            policy.set_all_vars(*[f[v.name][...]
-                                  for v in policy.all_variables])
-        return policy
-
-    # === Rollouts/training ===
-
-    def rollout(self, env, preprocessor, render=False, timestep_limit=None,
-                save_obs=False, random_stream=None):
-        """Do a rollout.
-
-        If random_stream is provided, the rollout will take noisy actions with
-        noise drawn from that stream. Otherwise, no action noise will be added.
-        """
-        env_timestep_limit = env.spec.tags.get("wrapper_config.TimeLimit"
-                                               ".max_episode_steps")
-        timestep_limit = (env_timestep_limit if timestep_limit is None
-                          else min(timestep_limit, env_timestep_limit))
-        rews = []
-        t = 0
-        if save_obs:
-            obs = []
-        ob = preprocessor.transform(env.reset())
-        for _ in range(timestep_limit):
-            ac = self.act(ob[None], random_stream=random_stream)[0]
-            if save_obs:
-                obs.append(ob)
-            ob, rew, done, _ = env.step(ac)
-            ob = preprocessor.transform(ob)
-            rews.append(rew)
-            t += 1
-            if render:
-                env.render()
-            if done:
-                break
-        rews = np.array(rews, dtype=np.float32)
-        if save_obs:
-            return rews, t, np.array(obs)
-        return rews, t
-
-    def act(self, ob, random_stream=None):
-        raise NotImplementedError
-
-    def set_trainable_flat(self, x):
-        self._setfromflat(x)
-
-    def get_trainable_flat(self):
-        return self._getflat()
-
-    @property
-    def needs_ob_stat(self):
-        raise NotImplementedError
-
-    def set_ob_stat(self, ob_mean, ob_std):
-        raise NotImplementedError
-
-
-def bins(x, dim, num_bins, name):
-    scores = U.dense(x, dim * num_bins, name, U.normc_initializer(0.01))
-    scores_nab = tf.reshape(scores, [-1, dim, num_bins])
-    return tf.argmax(scores_nab, 2)
-
-
-class GenericPolicy(Policy):
-    def _initialize(self, ob_space, ac_space, preprocessor, ac_noise_std):
-        self.ac_space = ac_space
-        self.ac_noise_std = ac_noise_std
+from ray.rllib.ppo.filter import NoFilter, MeanStdFilter
```

Comment:
TODO: move these to someplace common

Reply:
Yeah, good point! Let's do that as a follow up.